common/config.py

```
@@ -128,35 +130,51 @@ def get_llm_service(llm_config):
+except MilvusException as e:
+    embedding_store = None
+    service_status["embedding_store"] = {"status": "milvus error", "error": str(e)}
+except Exception as e:
+    embedding_store = None
+    service_status["embedding_store"] = {"status": "embedding error", "error": str(e)}
 
 support_collection_name = milvus_config.get("collection_name", "tg_support_documents")
 LogWriter.info(
     f"Setting up Milvus embedding store for SupportAI with collection_name: {support_collection_name}"
 )
 vertex_field = milvus_config.get("vertex_field", "vertex_id")
-support_ai_embedding_store = MilvusEmbeddingStore(
-    embedding_service,
-    host=milvus_config["host"],
-    port=milvus_config["port"],
-    support_ai_instance=True,
-    collection_name=support_collection_name,
-    username=milvus_config.get("username", ""),
-    password=milvus_config.get("password", ""),
-    vector_field=milvus_config.get("vector_field", "document_vector"),
-    text_field=milvus_config.get("text_field", "document_content"),
-    vertex_field=vertex_field,
-    alias=milvus_config.get("alias", "default"),
-)
+try:
+    support_ai_embedding_store = MilvusEmbeddingStore(
+        embedding_service,
+        host=milvus_config["host"],
+        port=milvus_config["port"],
+        support_ai_instance=True,
+        collection_name=support_collection_name,
+        username=milvus_config.get("username", ""),
+        password=milvus_config.get("password", ""),
+        vector_field=milvus_config.get("vector_field", "document_vector"),
+        text_field=milvus_config.get("text_field", "document_content"),
+        vertex_field=vertex_field,
+        alias=milvus_config.get("alias", "default"),
+    )
+    service_status["support_ai_embedding_store"] = {"status": "ok", "error": None}
+except MilvusException as e:
+    support_ai_embedding_store = None
+    service_status["support_ai_embedding_store"] = {"status": "milvus error", "error": str(e)}
```

Do we want to let CoPilot continue to run if it can't connect to an embedding store? My vote would be that CoPilot fails here. I agree we should capture this and have a useful error message so the user knows what to do.

---

That's a good question. It is requested by cloud, as when copilot crashes, the cloud controller will try to constantly restart the container. There is no way for it to check the health status (nor do they check the health status) and handle it gracefully unless someone go to the backend and read the logs.

---

But since this is not for cloud, we can raise the error and let it crash

---

Add the "raise" back so that copilot fails when it cannot connect to embedding store